spotlight-client/src/DataStore/TenantStore.ts

```diff
@@ -52,13 +50,16 @@ export default class TenantStore {
     // tenant mapped from domain should be locked
     const tenantFromDomain = getTenantFromDomain();
     if (tenantFromDomain) {
-      this.locked = true;
       this.currentTenantId = tenantFromDomain;
       // returning null renders an observable property immutable
       intercept(this, "currentTenantId", () => null);
     }
   }
 
+  get locked(): boolean {
+    return !!this.rootStore.userStore.stateCode || !!getTenantFromDomain();
```

Comment:
Not sure what you mean? We do want to lock the tenant in production based on the domain?

Reply:
This line does not make it super clear about what branch of logic users will take in the different environments. I think you should add a test to TenantStore.test for each branch of logic with a comment about when and where it is a possibility to document all of the possible branches for clarity.

-If the state_code exists (only true in the staging auth tenant for certain state users) the app is locked. 
-If the state_code is undefined (Recidiviz users, or production users), and the tenant cannot be found in the domain (which means it is staging), the app is not locked. 
-If the state_code is undefined (Recidiviz users, or production users), and the tenant CAN be found in the domain (which means it is production), the app is locked. 

Is that correct?

Reply:
Yes that should cover the cases. In production we turn off the auth wall so the state_code will always be undefined in that case. (No case of user state_code being defined in one state but domain is in another). I'm thinking of making all these cases clear in the tests and document how to configure all these things in the README?

Reply:
I think that a good comment on this line is better than in the README. Since this is where the logic lives, future developers may not know to go to the README to understand what these cases are.

Reply:
Added a comment explaining how `locked` is set + wrote an explanation in the README about locking users to a tenant in the staging environment.